### DISCUSSION TITLE
Allowing default window chrome to listen to Windows "Theme" preference

### DISCUSSION THREAD

**Discussion by ProbablePrime** [AvaloniaUI/Avalonia#10849] [Q&A] [answered]
I have windows set to "Dark":
![image](https://user-images.githubusercontent.com/8791132/228751035-21fdf032-27e4-4a66-abef-fb191ce1ca0d.png)

A default Avalonia window (with FluentTheme set to dark):
```
<Window xmlns="https://github.com/avaloniaui"
        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
        xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
        xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
        mc:Ignorable="d" d:DesignWidth="800" d:DesignHeight="450"
        x:Class="AccountDownloader.Views.AboutWindow"
        Title="AboutWindow">
  Welcome to Avalonia!
</Window>
```

Still renders with a "Light" themed title bar:
![image](https://user-images.githubusercontent.com/8791132/228751183-0f0e194c-5eba-47c7-bd27-38d7744d5fba.png)

Is there a way to make the window listen to the OS preferences of the user?

I'm on Windows 10 (21H2), some users in telegram indicated that this might not be possible in Windows 10 due to some out of date windows apis.

**Comment by timunie** [comment, marked as answer]
Please check out [FluentAvalonia](https://github.com/amwx/FluentAvalonia/). They provide a so called "core window" which also works on windows 10. If you don't want to use this library, you could still dig into the source code and see which Win32 hacks they did to achieve the desired result, see: https://github.com/amwx/FluentAvalonia/blob/master/src/FluentAvalonia.UI.Windowing/

Note: It's not only an Avalonia issue, see: https://stackoverflow.com/questions/57124243/winforms-dark-title-bar-on-windows-10 for example.

**Reply by ProbablePrime** [reply]
Thanks tim for the information, it helps!

**Comment by maxkatz6** [comment]
Aside from using third party controls or custom rendered title bar, it's an expected behavior on windows. 

Official and documentated API to change window decorations color was added only in windows 11 22000 - DWMWA_USE_IMMERSIVE_DARK_MODE attribute https://learn.microsoft.com/en-us/windows/win32/api/dwmapi/ne-dwmapi-dwmwindowattribute

There are undocumented APIs that had the same effect before, but we don't plan on using them for the older OS. 
Also, UWP apps specifically had an API for that even before, but classic win32 apps like wpf or avalonia can't use them.

**Comment by imvikysharma** [comment]
I suggest that you open a new issue in the Avalonia GitHub repository with detailed information about your system environment, including the version of Avalonia that you are using, the version of Windows 10 that you have installed, and any other relevant information. This will allow the developers to look into the issue and provide a more informed response.

I would like to recommend a book named "Chrome OS System Administrator's Booklet"

https://www.amazon.com/dp/B0BV7BF43B?maas=maas_adg_419F4F107F00519A5B1C275B372B05E5_afap_abs&ref_=aa_maas&tag=maas
